packages/mobile/src/account/CeloLite.tsx

```diff
@@ -1,62 +1,139 @@
 import SettingsSwitchItem from '@celo/react-components/components/SettingsSwitchItem'
+import TextButton from '@celo/react-components/components/TextButton'
 import colors from '@celo/react-components/styles/colors'
 import fontStyles from '@celo/react-components/styles/fonts'
 import * as React from 'react'
 import { WithNamespaces, withNamespaces } from 'react-i18next'
-import { ScrollView, StyleSheet, Text } from 'react-native'
+import { ScrollView, StyleSheet, Text, View } from 'react-native'
+import Modal from 'react-native-modal'
 import { connect } from 'react-redux'
 import i18n, { Namespaces } from 'src/i18n'
 import { headerWithBackButton } from 'src/navigator/Headers'
 import { RootState } from 'src/redux/reducers'
-import { setZeroSyncMode } from 'src/web3/actions'
+import { toggleZeroSyncMode } from 'src/web3/actions'
 
 interface StateProps {
   zeroSyncEnabled: boolean
+  gethStartedThisSession: boolean
 }
 
 interface DispatchProps {
-  setZeroSyncMode: typeof setZeroSyncMode
+  toggleZeroSyncMode: typeof toggleZeroSyncMode
 }
 
 type Props = StateProps & DispatchProps & WithNamespaces
 
 const mapDispatchToProps = {
-  setZeroSyncMode,
+  toggleZeroSyncMode,
 }
 
 const mapStateToProps = (state: RootState): StateProps => {
   return {
     zeroSyncEnabled: state.web3.zeroSyncMode,
+    gethStartedThisSession: state.web3.gethStartedThisSession,
   }
 }
 
-export class CeloLite extends React.Component<Props> {
+interface State {
+  modalVisible: boolean
+}
+
+export class CeloLite extends React.Component<Props, State> {
   static navigationOptions = () => ({
     ...headerWithBackButton,
     headerTitle: i18n.t('accountScreen10:celoLite'),
   })
 
+  state = {
+    modalVisible: false,
+  }
+
+  showModal = () => {
+    this.setState({ modalVisible: true })
+  }
+
+  hideModal = () => {
+    this.setState({ modalVisible: false })
+  }
+
+  onPressRestartModal = () => {
+    this.props.toggleZeroSyncMode(false)
```

Comment:
The naming has me a bit confused here. Any reason we dispatch this action here instead of restarting the app as the name says? I see below that the restart logic is in the saga instead, any reason we don't have it here?

Reply:
Good point. I updated the name to `onPressToggleWithRestartModal`. 

The restart logic is in the saga because we have to wait for geth to be turned on before restarting the app. Putting the restart logic here would require dispatching an action after the syncMode is done switching, and listening for that to know when to restart.

Reply:
👍